tests/framework/matcher/getter.go

```diff
@@ -23,15 +23,18 @@ func ThisPod(pod *v1.Pod) func() (*v1.Pod, error) {
 
 // ThisPodWith fetches the latest state of the pod based on namespace and name. If the object does not exist, nil is returned.
 func ThisPodWith(namespace string, name string) func() (*v1.Pod, error) {
-	return func() (p *v1.Pod, err error) {
+	return func() (*v1.Pod, error) {
```

Comment:
Let's add tests

Reply:
Not sure it's so simple. These functions uses a global client. Not sure it's possible to override the client. 

@xpivarc - any idea?

Reply:
it can be done by introducing getClient func pointer that will be override in the tests